python/ray/rllib/optimizers/aso_learner.py

```diff
@@ -43,18 +50,47 @@ def __init__(self, local_worker, minibatch_buffer_size, num_sgd_iter,
         self.stats = {}
         self.stopped = False
 
+
     def run(self):
         while not self.stopped:
             self.step()
 
     def step(self):
         with self.queue_timer:
-            batch, _ = self.minibatch_buffer.get()
+            # old_worker not None means that APPO agent is being used (the only agent that uses old_worker)
+            if self.old_worker:
+                batch, _ = self.minibatch_buffer.get()
+            else:
+                batch = self.inqueue.get()
```

Comment:
wait is this logic flipped..? by default there is no old_worker right? why does the behavior change then?

Reply:
The logic is "if appo, use minibatch buffer", "if impala, use inqueue". Empirically, I found impala to perform better with inqueue even when minibatch buffer == inqueue when num_sgd_iter =1 and minibatch_buffer_size =1